### ISSUE TITLE
Add 'strict' option for @include case sensitivity

### ISSUE THREAD

**Issue by mm-gmbd** [electricimp/Builder#35]
Filenames in `@includes` are not case sensitive, but the way they are registered as already-included or not *is*. This means that I can have the following:

```
@include once "some/path/file.nut"
@include once "some/path/File.nut"
```

And the "file.nut" file will be included twice.

**Comment by mm-gmbd** [comment]
Understood. Maybe a good middle ground would be to, in the event of filenames of this kind of "collision" (i.e. where two filenames are the same with the exception of casing), throw a warning? We have tried to be methodical in how we define shared files and libraries, which has lead to many many includes and having a method to catch/warn about this would be helpful.

**Comment by ppetrosh** [comment]
Even though I can see a lot of potential workarounds (even using some features of Builder itself), I believe there's something we can do about it. I'll give it a look (with a moderate priority though).

**Comment by ppetrosh** [comment]
A few thoughts on this:

* case sensitivity should be taken into account
* normalized (may be platform specific) path should be used for duplicate entries detection
* every type of resource: http/https, github, local file should be taken care of separately

I added the task to the backlog.

**Comment by rlipscombe** [comment]
> normalized (may be platform specific) path should be used for duplicate entries detection

Note: This is essentially impossible on Windows, because there's approximately a million different ways to refer to the "same" file.

See [this](https://stackoverflow.com/a/1816754/8446) (by me); or [this](http://pdh11.blogspot.co.uk/2009/05/pathcanonicalize-versus-what-it-says-on.html) (by Peter).

Best effort's fine, of course, but it's never going to be 100% reliable.

You _could_ hash the file, I guess...

**Comment by ppetrosh** [comment]
I would probably do this:

- hash file contents
- introduce a flag `--prohibit-duplicate-includes` that would raise an error in case one tries to include a file twice (regardless of the path, be it git and local file system for example)

**Comment by mm-gmbd** [comment]
Sounds like a good proposal to me 👍

**Comment by deldrid1** [comment]
+1 for supporting this

**Comment by betzrhodes** [comment]
Fixed in release [v2.6.0](https://github.com/electricimp/Builder/releases/tag/v2.6.0)
